File: scripts/extract_questions.py
#!/usr/bin/env python3
import re
import json

# Read the scenarios.ts file
with open('src/data/scenarios.ts', 'r', encoding='utf-8') as f:
    content = f.read()

questions = {}

# Pattern to match question blocks
# This regex captures: id, question text, and the entire options block
question_pattern = r"id:\s*['\"](q\d+-\d+)['\"],\s*type:\s*['\"][^'\"]+['\"],\s*question:\s*['\"]([^'\"]+)['\"],"

# Find all questions
for match in re.finditer(question_pattern, content):
    question_id = match.group(1)
    question_text = match.group(2)
    
    # Find the options block for this question
    # Look for the options array that follows this question
    start_pos = match.end()
    # Find the next question or end of questions array
    next_question_match = re.search(r"id:\s*['\"]q\d+-\d+['\"]", content[start_pos:])
    if next_question_match:
        options_block_end = start_pos + next_question_match.start()
    else:
        # Last question, find the end of questions array
        options_block_end = content.find('],', start_pos)
        if options_block_end == -1:
            options_block_end = len(content)
    
    options_block = content[start_pos:options_block_end]
    
    # Extract options
    options = {}
    option_pattern = r"id:\s*['\"](a\d+-\d+)['\"],\s*text:\s*['\"]([^'\"]+)['\"],\s*isCorrect:\s*(true|false),\s*explanation:\s*['\"]([^'\"]+)['\"]"
    
    for opt_match in re.finditer(option_pattern, options_block):
        opt_id = opt_match.group(1)
        opt_text = opt_match.group(2)
        opt_explanation = opt_match.group(4)
        
        options[opt_id] = {
            "text": opt_text,
            "explanation": opt_explanation
        }
    
    questions[question_id] = {
        "question": question_text,
        "options": options
    }

# Output as JSON
output = {"questions": questions}
print(json.dumps(output, indent=2, ensure_ascii=False))
print(f"\nExtracted {len(questions)} questions", file=__import__('sys').stderr)

File: scripts/extract-questions.js
/**
 * Script to extract questions from scenarios.ts and generate translation templates
 * Run with: node scripts/extract-questions.js
 */

const fs = require('fs');
const path = require('path');

// Read scenarios file
const scenariosPath = path.join(__dirname, '../src/data/scenarios.ts');
const scenariosContent = fs.readFileSync(scenariosPath, 'utf-8');

// Extract questions using regex
const questionRegex = /id:\s*['"](q\d+-\d+)['"],\s*type:\s*['"]([^'"]+)['"],\s*question:\s*['"]([^'"]+)['"],[\s\S]*?options:\s*\[([\s\S]*?)\]\s*\},/g;

const questions = {};
let match;

while ((match = questionRegex.exec(scenariosContent)) !== null) {
  const questionId = match[1];
  const questionText = match[3];
  const optionsText = match[4];
  
  // Extract options
  const optionRegex = /id:\s*['"]([^'"]+)['"],\s*text:\s*['"]([^'"]+)['"],\s*isCorrect:\s*(true|false),[\s\S]*?explanation:\s*['"]([^'"]+)['"]/g;
  const options = {};
  let optionMatch;
  
  while ((optionMatch = optionRegex.exec(optionsText)) !== null) {
    const optionId = optionMatch[1];
    const optionText = optionMatch[2];
    const explanation = optionMatch[4];
    
    options[optionId] = {
      text: optionText,
      explanation: explanation
    };
  }
  
  questions[questionId] = {
    question: questionText,
    options: options
  };
}

// Generate JSON structure
const questionsJson = JSON.stringify(questions, null, 2);

// Write to file
const outputPath = path.join(__dirname, '../messages/questions-template.json');
fs.writeFileSync(outputPath, questionsJson, 'utf-8');

console.log(`Extracted ${Object.keys(questions).length} questions`);
console.log(`Template saved to: ${outputPath}`);
console.log('\nYou can now copy this structure to your language files and translate the content.');

